Library/Homebrew/brew.sh

```diff
@@ -191,6 +191,16 @@ case "$HOMEBREW_COMMAND" in
   --config)    HOMEBREW_COMMAND="config" ;;
 esac
 
+if [[ -z "$HOMEBREW_DEVELOPER" ]]
+then
+  export HOMEBREW_GIT_CONFIG_FILE="$HOMEBREW_REPOSITORY/.git/config"
+  HOMEBREW_GIT_CONFIG_DEVELOPERMODE="$(git config --file="$HOMEBREW_GIT_CONFIG_FILE" --get homebrew.developermode)"
```

Comment:
> That's kinda the point of this PR

And that's what I don't think I understand.  I thought the point was to stop requiring HOMEBREW_DEVELOPER be set in order to use the dev-cmds (et al).  Why not just remove the "don't do anything unless `HOMEBREW_DEVELOPER` is set" guard on the dev-cmds (et al), instead of checking and setting things in a gitconfig to stealthily set HOMEBREW_DEVELOPER?